### ISSUE TITLE
[SOI] The Gitrog Monster not properly triggering on lands hitting grave consistently

### ISSUE THREAD

**Issue by dampete** [magefree/mage#1728]
The "Whenever one or more land cards are put into your graveyard from anywhere, draw a card" portion isn't triggering at any moment...

**Comment by drmDev** [comment]
thought there was another issue for this already open, but can't find it. Known issue - still unresolved. I'll see if I can dig up the related issue on it.

**Comment by drmDev** [comment]
The discussion on it must have been on reddit or the xmage forums. Either way I have tests added for this issue on an earlier commit:
97bfad470dcd5f6c08148b076d9028e304c7d9ad

The tests are failing since the card is still bugged and not triggering correctly.

**Comment by drmDev** [comment]
Related reddit post on issue:
https://www.reddit.com/r/XMage/comments/4ccqai/discarding_dakmor_salvage_doesnt_trigger_the/

**Comment by dampete** [comment]
Ok, this Reddit post only mentions Dredge (Library to Graveyard), but even as an effect of The Gitrog Monster's "sacrifice land" cost, when the land hits the Graveyard from Battlefield it doesn't trigger either.

PS.: Sorry if this is the incorrect way for reporting issues, I was looking to clone the repo and start working on some issues

**Comment by drmDev** [comment]
This is the correct way to report issues. It's the easiest way to keep track of what has been resolved and what hasn't.

Look forward to you joining in to help out!
Check out the getting started guide if you haven't already:
https://github.com/magefree/mage/wiki/Developer-Getting-Started

As well as the XMage Dev Talk forums:
http://www.slightlymagic.net/forum/viewforum.php?f=116

**Comment by dampete** [comment]
I started looking into the problem. Unfortunately I'm not a Java programmer, so it's taking a while to navigate through the code (for some reason "Go To Source/Implementation/etc" isn't working for me).

What I worked out until now is that triggers that are worded "when one or more X then Y" (meaning that the effect triggers only once no matter how many instances of X occur) all seem to be coded specifically, unlike "for each X then Y". I imagine using a code similar to Ulamog, the Infinite Gyre would make effects like Armageddon cause multiple triggers. Not sure why there aren't general events that can treat "when one or more" similar to how there are for single effects.

Anyway, in one of your links above, you commented about using ZONE_CHANGE instead of ZONE_CHANGE_GROUP. Since I can't navigate the code properly yet, I'm assuming ZONE_CHANGE_GROUP is the Event Type associated with a group of cards changing zone, as to only return True once for checkEventType (look at Sidisi, Brood Tyrant), so I assume that wouldn't fix the card and most probably make it work incorrectly.

PS.: Not sure if this is the right place to discuss this either

**Comment by drmDev** [comment]
For more indepth discussion on card implementation help, you'll want to use the XMage forums in the Dev Talk section. For instance, there's a thread there specifically on Shadows over Innistrad:
http://www.slightlymagic.net/forum/viewtopic.php?f=116&t=18413

**Comment by dampete** [comment]
Oh, I see... well, it seems that The Gitrog Monster is currently implemented, as per that thread. I'll close this issue then.

**Comment by drmDev** [comment]
The card is still bugged, re-opening.

**Comment by dampete** [comment]
Built and tested with current version. Here's some info:

I ran it with Sin Prodder and Evolving Wilds. If I put a card from top of library (Sin Prodder's trigger), The Gitrog Monster's ability triggers. But it doesn't trigger in either Battlefield to Graveyard (when I sac a land to Gitrog's cost and when I use Evolving Wilds ability), nor Hand to Graveyard (Discarding EOT)

**Comment by halljared** [comment]
This is a hallmark card of the new set. I'll look into this bug.

**Comment by dampete** [comment]
I did some investigating, might help:
http://www.slightlymagic.net/forum/viewtopic.php?f=116&t=18413&sid=c1995c536e6b637ac2749d8cc92c24bb&start=30#p194599

**Comment by halljared** [comment]
Ok looks good. Maybe worse than I was hoping but it's a start...
